src/chrome/utils/promises.ts

```diff
@@ -1,5 +1,22 @@
+import { asyncMap } from '../collections/async';
+
 /*---------------------------------------------------------
  * Copyright (C) Microsoft Corporation. All rights reserved.
  *--------------------------------------------------------*/
 
-export type PromiseOrNot<T> = Promise<T> | T;
\ No newline at end of file
+export type PromiseOrNot<T> = Promise<T> | T;
+
+/**
+ * Wait until the promise gets either resolved or rejected. If the promise gets rejected, waitForEnd will ignore that rejection and succeed anyways.
+ */
+export async function waitForEnd(...manyPromisesToWaitFor: Promise<unknown>[]): Promise<void> {
```

Comment:
This and `waitForSinglePromiseToEnd` make me nervous. Do we really want to be just swallowing exceptions with no logging? I've been bitten hard by stuff like this in the past where some method was swallowing an important exception and it was near impossible to figure out why something was failing.

At the very least, I think this is a place where the name should explicitly state that we are ignoring failures. Something like `awaitIgnoreFailures()` and `awaitAllIgnoreFailures()`, and log ignored exceptions to `warn`;

Reply:
You are right. That function could be easily misused.
I'll remove it so it's more difficult to ignore the failures of promises